docs/development/npm-package-development.md

```diff
@@ -0,0 +1,152 @@
+
+N/A
+
+## Guidance
+
+### Package Creation
+
+When starting the creation of a new package, it is recommended to use the `npm init` command instead of hand coding, to quickly and accuretly create a new package.json.  The results of accepting all the defaults, which can be done easily with this command `npm init -y`, is below:
+
+```
+{
+  "name": "package_name",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC"
+}
+```
+
+It should be noted that users and organizations can modify how `npm init` works, tailoring the resulting package.json to their needs.  For more information on this, check out the [official docs](https://docs.npmjs.com/cli/v9/commands/npm-init)
+
+#### package.json
+
+While **name** and **version** are the only fields that are required to publish a package, the team recommends a few others that provide more information to the user.
+
+* **description** - This is a string that will help users discover the package, especially if it is published to the public npm registry
+
+* **keywords** - This is an array of strings that will help users discover the package, especially if it is published
+to the public npm registry
+
+* **homepage** - A url to the project homepage,  most likely to point at a github repo.
+
+* **bugs** - The url to your project's issue tracker and / or the email address to which issues should be reported. These are helpful for people who encounter issues with your package.  Formatting should look like this:
+
+  ```
+    {
+      "url" : "https://github.com/owner/project/issues",
+      "email" : "project@hostname.com"
+    }
+  ```
+
+* **author** - An Author is one person or company who maintains the package.  This field usually consists of an object with a name, and optionally an email and url.
+
+* **contributors** - Contributors is an array of people who maintain the package
+
+* **license** - You should specify a license for your package so that people know how they are permitted to use it, and any restrictions you're placing on it.  More information on licenses and how they relate to the package.json can be [found here](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#license)
+
+* **files** - The list of files that you want published with your package.  This field is very handy when using a bundler to transpile code and you only want to include that transpiled code.  The team recommends not to include tests, which should reduce the package size, but to include docs.  For a listing of the files that are automatically included, check out the [npm docs here](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files)
+
+* **main** - This is the primary entry point to your package.  This should be a modul relative to the root of your package folder.  Most times this will be `index.js`, which is the default if this field is not set.
```

Comment:
we should have a section on export maps

Reply:
You can probably get away with just using import maps and having a fallback to `main`. The advantage of import maps is also that only those parts of the package are exposed, so users cannot import other files.

Reply:
The discussion in the last meeting was that it could make sense to land what we have and then as you two (@bennypowers, @jonkoops ) to PR in suggestions for the exports maps section. The alternative is @lholmquist will set up a short meeting with you to get your suggestions so he can do the follow on PR.  Does that make sense ?